prow/git/client.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Clients knows how to create clients for repos
+type Clients interface {
+	Repo(org, repo, dir string) (RepoClient, error)
+}
+
+// RepoClient exposes interactions with a git repo
+type RepoClient interface {
+	Publisher
+}
+
+func NewClients(host string, useSSH bool, user func() (name, email, login string, err error), token func() []byte, censor func(content []byte) []byte) Clients {
```

Comment:
If this client only ever uses existing clones of repos that should be made clear somewhere. Right now it isn't really clear to a consumer of the git package what the differences between the old and the new client are or which they should use.